### ISSUE TITLE
[bug]: fix accessibility violations fo `post-date-picker`

### ISSUE THREAD

**Issue by alionazherdetska** [swisspost/design-system#7620]
## 🐞 Describe the bug

Accessibility audit of the `post-date-picker` component revealed 3 violations. One is confirmed by all three tools (Axe, ARC Toolkit and Siteimprove browser extension) and must be fixed. Two additional violations are reported only by the Siteimprove browser extension and should be investigated for applicability.

---

### 1. 🔴 Input has no accessible name
> Confirmed by Axe, ARC Toolkit, Wave and Siteimprove · WCAG 4.1.2 Name, Role, Value (A)

```
Input has no accessible name
Description: The form control lacks an accessible name.
WCAG 2.2 Success Criterion:A [4.1.2 Name, Role, Value External link](https://www.w3.org/TR/WCAG22/#name-role-value)
<input type="text" class="form-control"
```

---

### 2. 🟡 Visible label and accessible name do not match
> Siteimprove only  · WCAG 2.5.3 Label in Name (A)

```
The accessible name of any interactive element should contain its visible text label.
Occurrences of this issue:
```

<img width="422" height="184" alt="Image" src="https://github.com/user-attachments/assets/6b10b061-aded-445d-97e8-b81e2b433380" />

---

### 3. 🟡 Role not inside the required context
> Siteimprove and AXE · WCAG 1.3.1 Info and Relationships (A)

```
This page element has a role that only makes sense in the context of another role.
```

<img width="419" height="382" alt="Image" src="https://github.com/user-attachments/assets/6a6ef2b7-571b-4dc9-9a22-f1c35411c61a" />

---

## 🛠️ To Reproduce

1. Open the `post-date-picker` story in Storybook **in the iframe** to avoid checks for Storybook-related markup
2. Run an accessibility audit using Axe, ARC Toolkit, or the Siteimprove browser extension
3. Observe the reported violations on the date input and calendar grid

**Comment by gfellerph** [comment]
Unblocking for progress, but take not of recent discussion in #7412
